src/main/java/com/snowflake/kafka/connector/internal/streaming/schemaevolution/iceberg/IcebergFieldNode.java

```diff
@@ -1,24 +1,83 @@
 package com.snowflake.kafka.connector.internal.streaming.schemaevolution.iceberg;
 
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
 import java.util.LinkedHashMap;
+import java.util.Map;
 import java.util.stream.Collectors;
 import org.apache.iceberg.types.Type;
 import org.apache.iceberg.types.Types;
 
 class IcebergFieldNode {
 
-  public final String name;
+  // todo consider refactoring into some more classes
+  private final IcebergColumnTypeMapper mapper = IcebergColumnTypeMapper.INSTANCE;
 
-  public final String snowflakeIcebergType;
+  final String name;
 
-  public final LinkedHashMap<String, IcebergFieldNode> children;
+  final String snowflakeIcebergType;
+
+  final LinkedHashMap<String, IcebergFieldNode> children;
 
   IcebergFieldNode(String name, Type apacheIcebergSchema) {
     this.name = name;
-    this.snowflakeIcebergType = IcebergColumnTypeMapper.mapToSnowflakeDataType(apacheIcebergSchema);
+    this.snowflakeIcebergType = mapper.mapToSnowflakeDataType(apacheIcebergSchema);
     this.children = produceChildren(apacheIcebergSchema);
   }
 
+  IcebergFieldNode(String name, JsonNode jsonNode) {
+    this.name = name;
+    this.snowflakeIcebergType = mapper.mapToColumnDataTypeFromJson(jsonNode);
+    this.children = produceChildren(jsonNode);
+  }
+
+  /**
+   * Method does not modify, delete any existing nodes and its types, names. It is meant only to add
+   * new children.
+   */
+  void merge(IcebergFieldNode modifiedNode) {
```

Comment:
nit: find a better name for modifiedNode. Maybe just `nodeToMerge`? Not an easy one to name.